src/Types/Transaction.purs

```diff
@@ -20,27 +29,129 @@ newtype TxBody = TxBody
     outputs :: Array TransactionOutput,
     fee :: Coin,
     ttl :: Maybe Slot,
-    certs :: Maybe Unit, -- Certificates,
-    withdrawals :: Maybe Unit, -- Withdrawals,
-    update :: Maybe Unit, -- Update,
-    auxiliary_data_hash :: Maybe String, -- AuxiliaryDataHash, - script hashes
+    certs :: Maybe (Array Certificate),
+    withdrawals :: Maybe (Map RewardAddress Coin),
+    update :: Maybe Update,
+    auxiliary_data_hash :: Maybe AuxiliaryDataHash, 
     validity_start_interval :: Maybe Slot,
-    mint :: Maybe Value, -- Mint
-    script_data_hash :: Maybe String, -- ScriptDataHash,
+    mint :: Maybe Mint,
+    script_data_hash :: Maybe ScriptDataHash,
     collateral :: Maybe (Array TransactionInput),
     required_signers :: Maybe (Array RequiredSigner),
     network_id :: Maybe NetworkId
 }
 
+newtype ScriptDataHash = ScriptDataHash String
+
+newtype Mint = Mint Value
+
+newtype AuxiliaryDataHash = AuxiliaryDataHash String
+
+type Update = 
+  { proposed_protocol_parameter_updates :: ProposedProtocolParameterUpdates
+  , epoch :: Epoch
+  }
+
+newtype ProposedProtocolParameterUpdates 
+  = ProposedProtocolParameterUpdates (Map GenesisHash ProtocolParamUpdate)
+
+newtype GenesisHash = GenesisHash String
+
+type ProtocolParamUpdate = 
+  { minfee_a :: Maybe Coin,
+    minfee_b :: Maybe Coin,
+    max_block_body_size :: Maybe Int,
+    max_tx_size :: Maybe Int,
+    max_block_header_size :: Maybe Int,
+    key_deposit :: Maybe Coin,
+    pool_deposit :: Maybe Coin,
+    max_epoch :: Maybe Epoch,
+    n_opt :: Maybe Int,
+    pool_pledge_influence :: Maybe Rational,
+    expansion_rate :: Maybe UnitInterval,
+    treasury_growth_rate :: Maybe UnitInterval,
+    d :: Maybe UnitInterval,
+    extra_entropy :: Maybe Nonce,
+    protocol_version :: Maybe (Array ProtocolVersion),
+    min_pool_cost :: Maybe Coin,
+    ada_per_utxo_byte :: Maybe Coin,
+    cost_models :: Maybe Costmdls,
+    execution_costs :: Maybe ExUnitPrices,
+    max_tx_ex_units :: Maybe ExUnits,
+    max_block_ex_units :: Maybe ExUnits,
+    max_value_size :: Maybe Int
+  }
+
+type ExUnitPrices =
+  { mem_price :: SubCoin
+  , step_price :: SubCoin
+  }
+
+type ExUnits =
+  { mem :: BigInt.BigInt
+  , steps :: BigInt.BigInt
+  }
+
+type SubCoin = UnitInterval
+
+type RewardAddress = 
+  { network :: UInt
+  , payment :: StakeCredential
+  }
+
+data StakeCredential
+  = Key Ed25519KeyHash
+  | Script ScriptHash
+
+newtype Ed25519KeyHash = Ed25519KeyHash String 
+
+newtype ScriptHash = ScriptHash String
+
+newtype Costmdls = Costmdls (Map Language CostModel)
+
+data Language = PlutusV1
+
+newtype CostModel = CostModel (Array Int)
+
+type ProtocolVersion =
```

Comment:
Would it make sense to use `UInt`s for these fields? Assuming we want to track the serialization-lib's representation as closely as possible